boa/src/realm.rs

```diff
@@ -32,26 +32,31 @@ impl Realm {
         let _timer = BoaProfiler::global().start_event("Realm::create", "realm");
         // Create brand new global object
         // Global has no prototype to pass None to new_obj
-        let global = Value::from(Object::default());
+        let mut global = Object::default();
 
         // Allow identification of the global object easily
-        global.set_data(crate::object::ObjectData::Global);
+        global.data = ObjectData::Global;
```

Comment:
You can add attribute `#[allow(clippy::field_reassign_with_default)]` on the `create` method to make `clippy` happy.

Reply:
Does clippy offer a better suggestion?

Reply:
The code was triggering a clippy lint, but it seems to be a false positive, the code proposed by clippy wasn't valid.